parsec/core/fs/remote_loader.py

```diff
@@ -351,15 +356,41 @@ def __init__(
         self.local_storage = local_storage
 
     async def load_blocks(self, accesses: List[BlockAccess]) -> None:
+        async with open_service_nursery() as nursery:
+            async with await self.receive_load_blocks(accesses, nursery) as receive_channel:
+                async for value in receive_channel:
+                    pass
+
+    async def receive_load_blocks(
+        self, blocks: List[BlockAccess], nursery: trio.Nursery
+    ) -> "MemoryReceiveChannel[BlockAccess]":
         """
-        Raises:
-            FSError
-            FSRemoteBlockNotFound
-            FSBackendOfflineError
-            FSWorkspaceInMaintenance
-        """
-        for access in accesses:
-            await self.load_block(access)
+           Raises:
+               FSError
+               FSRemoteBlockNotFound
+               FSBackendOfflineError
+               FSWorkspaceInMaintenance
+           """
+        blocks_iter = iter(blocks)
+        channels: Tuple[
+            "MemorySendChannel[BlockAccess]", "MemoryReceiveChannel[BlockAccess]"
+        ] = open_memory_channel(0)
+        send_channel, receive_channel = channels
+
+        async def _loader(send_channel: "MemorySendChannel[BlockAccess]") -> None:
```

Comment:
Why is the `send_channel` type in quotes?

Reply:
Mypy assumes it's a generic type but the interpreter raise an error, to fix that we put the type in quotes.
See https://stackoverflow.com/questions/55076778/why-isnt-this-function-type-annotated-correctly-error-missing-type-parameters for more info